### PR TITLE
Fix aggregated pull requests with multi project/working dirs

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---
Based on: https://github.com/jfrog/froggit-go/pull/98
- Fix using aggregated fixing pull request with multiple projects/workings dirs
- Refactor create-fix-pull-request
- Added tests

### CODE REVIEW

File: commands/createfixpullrequests_test.go [hunk 1]
```diff
@@ -158,12 +171,16 @@ func TestCreateFixPullRequestsCmd_Run(t *testing.T) {
...
+				resultDiff, err := verifyDependencyFileDiff("main", branch, test.packageDescriptorPaths...)
+				assert.NoError(t, err)
+				assert.Contains(t, test.expectedDiff, string(resultDiff))
+			}
 			// Defers
-			restoreEnv()
-			server.Close()
+			defer func() {
+				restoreEnv()
+				server.Close()
```
Comment (sverdlov93) [inline commands/createfixpullrequests_test.go:182]:
```suggestion
				assert.NoError(server.Close())
```
and same for all err func on this file

Comment (sverdlov93) [reply commands/createfixpullrequests_test.go:182]: defer cleanUp() >>

defer func(){
    assert.NoError(clenaup())
}

File: commands/createfixpullrequests.go [hunk 2]
```diff
@@ -54,41 +57,64 @@ func (cfp *CreateFixPullRequestsCmd) Run(configAggregator utils.RepoAggregator,
...
 
-func (cfp *CreateFixPullRequestsCmd) scanAndFixRepository(repository *utils.Repository, branch string, client vcsclient.VcsClient) error {
-	baseWd, err := os.Getwd()
+func (cfp *CreateFixPullRequestsCmd) scanAndFixRepository(repository *utils.Repository, branch string, client vcsclient.VcsClient) (err error) {
+	cfp.baseWd, err = os.Getwd()
 	if err != nil {
-		return err
+		return
 	}
+	cfp.setCommandPreRequisites(repository, branch, client)
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/createfixpullrequests.go:65]: done

File: commands/createfixpullrequests.go [hunk 3]
```diff
@@ -54,41 +57,64 @@ func (cfp *CreateFixPullRequestsCmd) Run(configAggregator utils.RepoAggregator,
...
+		cfp.details.Project = &repository.Projects[i]
+		cfp.projectTech = ""
+		if err = cfp.scanAndFixProject(repository); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (cfp *CreateFixPullRequestsCmd) setCommandPreRequisites(repository *utils.Repository, branch string, client vcsclient.VcsClient) {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/createfixpullrequests.go:76]: done

File: commands/testdata/createfixpullrequests/aggregate-multi-project/.frogbot/frogbot-config.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+      branches:
+        - main
+      repoName: "aggregate-multi-dir"
+    scan:
+      projects:
+        - workingDirs:
+            - npm
+        - workingDirs:
+            - pip
+          pipRequirementsFile: requirements.txt
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/testdata/createfixpullrequests/aggregate-multi-project/.frogbot/frogbot-config.yml:12]: yes

File: commands/utils/utils.go [hunk 5]
```diff
@@ -172,11 +172,11 @@ func Md5Hash(values ...string) (string, error) {
...
 // Generates MD5Hash from a vulnerabilityDetails
 // The map can be returned in different order from Xray, so we need to sort the strings before hashing.
-func VulnerabilityDetailsToMD5Hash(vulnerabilityDetails map[string]*VulnerabilityDetails) (string, error) {
+func VulnerabilityDetailsToMD5Hash(vulnerabilities ...*VulnerabilityDetails) (string, error) {
 	h := crypto.MD5.New()
-	keys := make([]string, 0, len(vulnerabilityDetails))
-	for k, v := range vulnerabilityDetails {
-		keys = append(keys, k+v.FixVersion)
+	var keys []string
+	for _, v := range vulnerabilities {
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/utils/utils.go:178]: done

File: commands/utils/utils.go [hunk 6]
```diff
@@ -172,11 +172,11 @@ func Md5Hash(values ...string) (string, error) {
 
 // Generates MD5Hash from a vulnerabilityDetails
 // The map can be returned in different order from Xray, so we need to sort the strings before hashing.
-func VulnerabilityDetailsToMD5Hash(vulnerabilityDetails map[string]*VulnerabilityDetails) (string, error) {
+func VulnerabilityDetailsToMD5Hash(vulnerabilities ...*VulnerabilityDetails) (string, error) {
 	h := crypto.MD5.New()
```
[the remark this answers is not in the document]
Comment (omerzi) [reply commands/utils/utils.go:176]: done

File: commands/createfixpullrequests_test.go (hunk 1 above)
Comment (omerzi) [reply commands/createfixpullrequests_test.go:182]: these function does not return errors

**Pull request merged**